commitlint.config.ts

```diff
@@ -428,6 +432,18 @@ module.exports = {
                     ];
                 },
 
+                'too-many-spaces': ({raw}: {raw:any}) => {
+                    let rawStr = convertAnyToString(raw, "raw");
+                    rawStr = removeAllCodeBlocks(rawStr);
+                    let offence = (rawStr.match(`[^.]  `) !== null);
+                    // && (rawStr.match(`\\.   `) === null)
```

Comment:
@realmarv this line above is a bad practice; I thought about explaining it here to you but I ended up rather explaining it in our WorkflowGuidelines document; please read it there and ask me if you have any questions

Reply:
@realmarv and BTW whenever you want to start coding in the branch for this PR again, remember to do git pull first because I added 2 commits myself

Reply:
> @realmarv this line above is a bad practice; I thought about explaining it here to you but I ended up rather explaining it in our WorkflowGuidelines document; please read it there and ask me if you have any questions

Do you mean removing those committed lines?

Reply:
removing is one way of fixing the bad practice

Reply:
This case `(rawStr.match('\\.   ') === null) `is automatically handled by` (rawStr.match('[^.]  ') !== null)`. Because when there are 3 spaces somewhere the last two spaces will always match this pattern `'[^.]  '` . That's why I commented it. But I'm not sure if I should explain it in a comment or just remove the commented line.

Reply:
> But I'm not sure if I should explain it in a comment or just remove the commented line.

Both options are valid, but leaving the comment there without explanation is the bad practice.